app/javascript/components/greeting.js

```diff
@@ -0,0 +1,22 @@
+import React, { useState, useEffect } from 'react';
+
+const Greeting = () => {
+  const [greeting, setGreeting] = useState('Loading...');
+
+  useEffect(() => {
+    // Fetch random greeting from the API
+    fetch('/api/greetings/random')
+      .then(response => response.json())
+      .then(data => setGreeting(data.greeting))
+      .catch(error => console.error('Error fetching greeting:', error));
+  }, []);
```

Comment:
- [ ] I see you are fetching the data directly, however, you will notice the requirements says you should add a reducer, action and store. This means you should use redux for the state management. 
![image](https://github.com/geekelo/hello-rails-react/assets/86200356/0f329146-1a95-4a31-95bc-509dea8e5879)